### DISCUSSION TITLE
Feature Request : Implement a API

### DISCUSSION THREAD

**Discussion by Sellto** [linuxserver/Heimdall#696] [Ideas]
Hello,

I work with my team on a framework for Big Data analysis : [FADI](https://github.com/cetic/fadi). This toolkit is mainly based on Helm and Kubernetes. We want to integrate a dashboard that would make navigation between services (Ingress) more interactive and pleasant. Heimdall seems to be the ideal candidate.

However, we would like to be able to automate the tasks of adding elements and users so that the dashboards are self-generated according to the services present in our cluster of containerized applications. At the moment this is not possible using only the routes in the file `routes/web.php`. 
We hope this use case will lead you to develop the API of your application. Because we prefer to use an existing mature tool and support it rather than developing it from scratch.

With this in mind, if this functionality were to come to light, we certainly plan to collaborate to enrich the list of enhanced supported applications and transmit a lot of feedback.

**Comment by KodeStar** [comment]
It's certainly possible, probably wouldn't even be that difficult. Just needs some routes adding to routes/api.php. Probably the hardest part would be the auth side.

**Comment by KodeStar** [comment]
This should be coming in v3 which is currently in development

**Comment by Vad1mo** [comment]
When is V3 planned to arrive?

**Comment by aptalca** [comment]
You must be new on the internet 😄

**Comment by KodeStar** [comment]
I've started implementing this in 2.5.0, if you are still interested in the feature please let me know what you would like to see in it - https://github.com/linuxserver/Heimdall/projects/5
